### PR TITLE
chore(metrics): add input_buffer_blocking_duration metrics

### PR DESCRIPTION
**Pull request by KeXiangWang**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

Add metrics for input's blocking time.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* Other (please specify in the release note below)
Metrics

### Release note

Please create a release note for your changes. In the release note, focus on the impact on users, and mention the environment or conditions where the impact may occur.

### CODE REVIEW

**Comment by KeXiangWang** [comment]
One limitation of this PR is that the metrics may not be able to work for `join` operator, as one `join` have to `merge` operator as input. Two `merge` operators can be waiting at the same time. So the waiting ratio(waiting time / total time) can be more than 1. 

I think we may need to deal with merge as a special case.

Any suggestions are welcomed.

File: src/stream/src/executor/merge.rs [hunk 1]
```diff
@@ -35,6 +35,9 @@ pub struct MergeExecutor {
     /// Belonged actor id.
     actor_id: ActorId,
 
+    /// Actor id as String
+    actor_id_str: String,
```
Comment (BugenZhao) [inline src/stream/src/executor/merge.rs:39]: No need to put it in `self`? 🤔

**Review by BugenZhao — COMMENTED** [review]
LGTM

**Comment by BugenZhao** [comment]
> One limitation of this PR is that the metrics may not be able to work for `join` operator, as one `join` have to `merge` operator as input. Two `merge` operators can be waiting at the same time. So the waiting ratio(waiting time / total time) can be more than 1.
> 
> I think we may need to deal with merge as a special case: add another metrics specifically for `join`.
> 
> Any suggestions are welcomed.

Also take the executor id of the Merge into account?

**Comment by skyzh** [comment]
Isn't this backpressure rate?

**Comment by skyzh** [comment]
Okay get it. Backpressure rate is for actor level. This is for merge executor.

**Comment by KeXiangWang** [comment]
> Isn't this backpressure rate?

My opinion: 
Backpressure is to detect whether this actor is blocked by downstream actors.

This is more like "fore pressure", which is used to detect whether this actor is blocked by upstream actors.
Imagine we have a high-load and low-throughput actor A and low-load and high-throughput actor B. And A is B's upstream actor. Then B could be blocked A. The merge executor in B will have to wait for A's output. This PR is to detect how much time  B actor will wait A.

File: src/stream/src/executor/merge.rs (hunk 1 above)
Comment (BugenZhao) [reply src/stream/src/executor/merge.rs:39]: The `execute_inner` is also only be called once, so it's okay to make it a local variable. :)

Comment (KeXiangWang) [reply src/stream/src/executor/merge.rs:39]: Ah, you are right!

File: src/stream/src/executor/receiver.rs [hunk 2]
```diff
@@ -81,9 +87,18 @@ impl Executor for ReceiverExecutor {
         let mut status = self.status;
         let metrics = self.metrics.clone();
         let actor_id_str = self.actor_id.to_string();
+        let upstream_fragment_id_str = self.upstream_fragment_id.to_string();
+        let mut start_time = minstant::Instant::now();
```
Comment (BugenZhao) [inline src/stream/src/executor/receiver.rs:91]: The start time for the first message seems incorrect here: it'll record before we start to poll the stream of this executor. Maybe we need to rewrite it with `futures_async_stream`.

**Review by BugenZhao — APPROVED** [review]

**Pull request merged**